src/main/groovy/com/ofg/pipeline/core/PipelineBuilder.groovy

```diff
@@ -27,7 +27,8 @@ class PipelineBuilder<P extends Project> {
         stageContext.with(closure)
     }
 
-    private class StageContext {
+    //Cannot be private as an external closure is called in its context
+    class StageContext {
```

Comment:
Good idea. I've tried it in the morning, but it doesn't want to fail when code is in jenkins-pipeline-dsl project itself. The same test case fails when run against the same codebase, however, packaged as a jar... I will try to return to that later.